### PR TITLE
Demand changes on alter user

### LINKED ISSUE #71227: Meaningless query `ALTER USER default` is accepted
It accepts the ALTER USER query with an empty list of modifications.

### PR DESCRIPTION
**Pull request by arthurpassos**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixes https://github.com/ClickHouse/ClickHouse/issues/71227

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Parsers/Access/ParserCreateUserQuery.cpp [hunk 1]
```diff
@@ -655,6 +657,13 @@ bool ParserCreateUserQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
     else if (alter)
         names->concatParts();
 
+    bool alter_command_with_no_changes = alter && pos_after_parsing_names == pos;
```
Comment (evillique) [inline src/Parsers/Access/ParserCreateUserQuery.cpp:660]: I feel like `alter_query_with_no_changes` would be better here

File: tests/queries/0_stateless/03254_test_alter_user_no_changes.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+-- Tags: no-parallel
+
+create user u_03254_alter_user;
+alter user u_03254_alter_user; -- { clientError SYNTAX_ERROR }
```
Comment (evillique) [inline tests/queries/0_stateless/03254_test_alter_user_no_changes.sql:4]: Probably a good idea to drop it after that

**Comment by arthurpassos** [comment]
Done, can you enable CI?

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit ffc88fd8a32fc183605d409c1028d865adca6cef with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71286/ffc88fd8a32fc183605d409c1028d865adca6cef/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/d14ca872d446fd5d1f6263860e2743ee6d7cd987/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by arthurpassos** [comment]
test_scheduler/test.py::test_workload_entity_keeper_storage	 - https://github.com/ClickHouse/ClickHouse/pull/71325

01290_max_execution_speed_distributed - https://github.com/ClickHouse/ClickHouse/issues/71366

**Comment by arthurpassos** [comment]
@evillique can you merge it?

**Comment by evillique** [comment]
@arthurpassos Can you merge the fresh master into this branch?

**Comment by arthurpassos** [comment]
> @arthurpassos could you please merge the fresh master into this branch?

I did, but this ref does not seem any better

**Comment by arthurpassos** [comment]
@evillique should I merge it again?

**Comment by clickgapai** [comment]
Hi @arthurpassos @evillique — while reviewing this PR I found the following:

- **Bug (P2):** `ALTER USER` no-op detection bypassed by `ON CLUSTER` and `IN storage_name` clauses — https://github.com/ClickHouse/ClickHouse/issues/104677

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**